chrombpnet/helpers/preprocessing/auto_shift_detect.py

```diff
@@ -35,6 +37,27 @@ def convolve(to_scan, longer_seq):
     for i in range(len(longer_seq) - len(to_scan) + 1):
         vals.append(np.sum(to_scan*longer_seq[i:i+len(to_scan)]))
     return vals
+    
+def stream_filtered_tagaligns(src_tagaligns_stream, genome_file, out_stream):
+    '''
+    Given a tagalign subprocess stream and reference genome file, filters
+    out any reads in chromosomes not included in the reference. Reads in the
+    reference chromosomes are sent to the specified output stream.
+    
+    Returns:
+        Boolean. Indicates whether any reads not in the reference fasta were 
+        detected.
+    '''
+    has_unknown_chroms = False
+    with pyfaidx.Fasta(genome_file) as g:
+        for line in iter(src_tagaligns_stream.stdout.readline, b''):
+            tagalign_chrom = line.decode('utf-8').strip().split('\t')[0]
+            if tagalign_chrom in g.keys():
+                out_stream.stdin.write(line)
```

Comment:
@riyavsinha is this writing to the `/tmp/` directory?

Reply:
A user reported an error at this line ... `BrokenPipeError: [Errno 32] Broken pipe` and `sort: write failed: /tmp/sortU7nWxt: No space left on device`

Reply:
Exact command they used if helpful - this is on the tutorial data and run on Sherlock 

`sbatch -p gpu,owners --gpus 1 -c 10 --mem 120GB --time 1-0 --job-name=${job_header} --output=${job_header}.out --error=${job_header}.err --wrap singularity exec --contain --nv -e --no-mount hostfs /oak/stanford/groups/wjg/skim/software/sing_containers/chrombpnet.sif chrombpnet pipeline -ibam ${bam} -d ${mode} -g ${genome} -c ${chromsize} -p ${peaks} -n ${nonpeaks} -fl ${fold} -b ${biasmodel} -o ${outfile}`

Reply:
`No space left on device` likely means they don't have disk space?

Reply:
They have sufficient space not the disk they are running it. Wondering why this command writes to the `/tmp/` folder and how much we can write to this folder on Sherlock.

Reply:
Error seems to be with sorting though. The file `/tmp/sortU7nWxt` seems to be a unix temp file.

Reply:
Changing the `TMPDIR` environment variable may be an option.

Reply:
This specific PR does not increase file or memory usage as it only introduces an extra filtering step which works with streams.